core/metacling/src/TCling.cxx

```diff
@@ -1890,7 +1900,9 @@ void TCling::RegisterModule(const char* modulename,
       }
    }
 
-   if (gIgnoredPCMNames.find(modulename) == gIgnoredPCMNames.end()) {
+   // Don't do "PCM" optimization with runtime modules because we are loading libraries at decl deserialization time and
+   // it triggers infinite deserialization chain.
+   if (!hasCxxModule && gIgnoredPCMNames.find(modulename) == gIgnoredPCMNames.end()) {
```

Comment:
What's the point of finding module names if we don't have modules?

Reply:
This "modulename" seems to represent dictionary name and I think it's possible that pch takes this branch.

Reply:
This is the set containing all empty root pcms.